### PR TITLE
Updated Documentation

### PR DESCRIPTION
**Pull request by Abdullahshah**
Version 2 (current):
The api-docs now uses a new recursive way to find source code and build summary pages for them for our documentation. The only change moving forward would be to add a quick description in the `__init__.py` files for each module. An example would be:
`
__init__.py:
"""
A quick description of the module
"""`

Sphinx auto-doc will take this string and set it as a quick description for the module's page.

To generate the documentation, navigate to `eva/api-docs/` and run `make clean && make html`. This will create the HTML files under `_build` which you can use to view the changes locally. It also generates the `.rst` files under `api-docs/_autosummary` which are used by Spinx to generate the HTML content

Version 1 (OLD):
Created script `generate_documentation.py` which creates the documenation `.rst` using the jinja templating engine. 
(this script can only be run from the root directory -> but there is a check for this in the script)

To test these changes, enter the conda environment, navigate to the directory `api-docs` and run `make html`. This will generate a local copy of the website under the directory `_build` which can be opened in the browser.

### CODE REVIEW

**Review by gaurav274 — CHANGES_REQUESTED** [review]
Please also add the installation dependencies to build using sphinx .

File: api-docs/generate_documentation.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+
+# Generate folder names and file names #
+
+folders = []
+for (dirpath, dirnames, filenames) in walk('src'):
+    folders.extend(dirnames)
+    break
+
+if "__pycache__" in folders:
```
Comment (gaurav274) [inline api-docs/generate_documentation.py:21]: Make a list of ignore_folders

File: api-docs/generate_documentation.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+}
+
+"""
+files_per_folder = {}
+for folder in folders:
+
+    files_per_folder[folder] = {}
+
+    subfolders = []
+    for (dirpath, dirnames, filenames) in walk('src/' + folder):
```
Comment (gaurav274) [inline api-docs/generate_documentation.py:64]: Move line number 64-70 to function

File: api-docs/generate_documentation.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+            print("Writing documentation sub rst file for",
+                  folder + "." + sub_folder_name,
+                  "at:", sub_rst_path)
+            with open(sub_rst_path, "w") as fh:
+                fh.write(rendered_sub_folder_rst_template)
+    else:
+        path = os.getcwd() + "/api-docs/documentation/" + folder + ".rst"
+        print("Writing documentation rst file for", folder, "at:", path)
+        with open(path, "w") as fh:
+            fh.write(rendered_main_rst_template)
```
Comment (gaurav274) [inline api-docs/generate_documentation.py:187]: Modularize this script

File: api-docs/generate_documentation.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        subpackages = []
+
+    # title_name converts "catalog" to "Catalog"
+    rendered_main_rst_template = main_template.render(
+        title_name=folder[0].upper() + folder[1:],
+        folder=folder,
+        files=sub_folders['root'],
+        subpackages=subpackages)
+
+    if createDocsSubFolder:
```
Comment (gaurav274) [inline api-docs/generate_documentation.py:156]: Make it generic enough to handle nested directories

File: api-docs/documentation_old/catalog/catalog.models.rst [hunk 5]
```diff
@@ -0,0 +1,38 @@
+catalog.models package
```
[the remark this answers is not in the document]
Comment (Abdullahshah) [reply api-docs/documentation_old/catalog/catalog.models.rst]: Removed both.

**Comment by Abdullahshah** [comment]
No longer need generate_documentation.py. Deprecating that script for now but may need it later if api-docs needs further changes.

**Review by Abdullahshah — COMMENTED** [review]
Check the Conversation tab for updates to documentation process.

**Comment by gaurav274** [comment]
@Abdullahshah  Any updates on this PR? Let's merge this as early as possible.

**Comment by xzdandy** [comment]
Add sphinx to conda environment file.

**Review by gaurav274 — APPROVED** [review]
LGTM

**Pull request merged**